addons/api/addon/transforms/object.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Transform from '@ember-data/serializer/transform';
+import { TrackedObject } from 'tracked-built-ins';
+
+export default class ObjectTransform extends Transform {
+  /**
+   * @param serialized
+   * @returns {object}
+   */
+  deserialize(serialized) {
+    if (Object.keys(serialized).length) {
```

Comment:
Won't this error out if it's `undefined`?

And shouldn't we still return a tracked object even if it is?

Reply:
good catch, it will error out, I'll add a fix for that in a separate PR since this is merged. 

I wasn't sure if that's needed when handling an empty object, do you think we would still need to track it?

Reply:
Yeah I think so, if the API didn't return anything but a user is updating it we still need it to be tracked when we add to the object